Mathlib/RingTheory/FilteredAlgebra/AssociatedGraded.lean

```diff
@@ -0,0 +1,166 @@
+    {n : ι} (hn : n ∉ s) : mk F F_lt s f n = 0 := by
+  dsimp only [Finset.coe_sort_coe, mk, AddMonoidHom.coe_mk, ZeroHom.coe_mk,
+    DFinsupp.mk_apply, DirectSum.mk]
+  rw [dif_neg hn]
+
+section support
+
+noncomputable instance : ∀ (i : ι) (x : GradedPiece F F_lt i), Decidable (x ≠ 0) :=
+  fun _ x ↦ Classical.propDecidable (x ≠ 0)
+
+@[simp]
+theorem support_of (i : ι) (x : GradedPiece F F_lt i) (h : x ≠ 0) : (of x).support = {i} :=
+  DFinsupp.support_single_ne_zero h
+
+theorem support_of_subset {i : ι} {b : GradedPiece F F_lt i} : (of b).support ⊆ {i} :=
+  DFinsupp.support_single_subset
+
+theorem sum_support_of (x : AssociatedGraded F F_lt) : (∑ i ∈ x.support, of (x i)) = x :=
+  DFinsupp.sum_single
+
+end support
+
+theorem sum_univ_of [Fintype ι] (x : AssociatedGraded F F_lt) :
+    ∑ i ∈ Finset.univ, of (x i) = x := by
+  apply DFinsupp.ext (fun i ↦ ?_)
+  rw [DFinsupp.finset_sum_apply]
+  simp [of_apply]
+
+theorem mk_injective (s : Finset ι) : Function.Injective (mk F F_lt s) :=
+  DFinsupp.mk_injective s
+
+theorem of_injective (i : ι) : Function.Injective (of (i := i) (F := F) (F_lt := F_lt)) :=
+  DFinsupp.single_injective
+
+end AssociatedGraded
+
+open AddSubgroup
+
+namespace GradedPiece
+
+/-- Obtaining an element of `GradedPiece i` from an element of `F i`. -/
+def mk {i : ι} : (ofClass (F i)) →+ GradedPiece F F_lt i :=
+  QuotientAddGroup.mk' ((ofClass (F_lt i)).addSubgroupOf (ofClass (F i)))
+
+section
+
+lemma induction_on {i : ι} {C : GradedPiece F F_lt i → Prop} (x : GradedPiece F F_lt i)
+    (H : ∀ z, C (GradedPiece.mk F F_lt z)) : C x :=
+  QuotientAddGroup.induction_on x H
+
+lemma mk_eq {i : ι} (x : F i) : mk F F_lt x = ⟦x⟧ := rfl
+
+lemma HEq_rfl {i j : ι} {r : A} (h : i = j) (hi : r ∈ ofClass (F i)) (hj : r ∈ ofClass (F j)) :
+    HEq (mk F F_lt ⟨r, hi⟩) (mk F F_lt ⟨r, hj⟩) :=
+  h ▸ HEq.rfl
+
+lemma HEq_eq_mk_eq {i j : ι} {x : GradedPiece F F_lt i} {y : GradedPiece F F_lt j} {r s : A}
+    (h : i = j) (e : r = s) (hi : r ∈ ofClass (F i)) (hj : s ∈ ofClass (F j))
+    (hx : x = mk F F_lt ⟨r, hi⟩) (hy : y = mk F F_lt ⟨s, hj⟩) : HEq x y := by
+  rw [hx, hy]
+  subst e
+  exact HEq_rfl F F_lt h hi hj
+
+-- Will be easier to use if HMul intances for F i is added and some other refactor is done.
+lemma HEq_eq_mk_coe_eq {i j : ι} {x : GradedPiece F F_lt i} {y : GradedPiece F F_lt j}
+    (r : ofClass (F i)) (s : ofClass (F j)) (h : i = j) (e : (r : A) = (s : A))
+    (hx : x = mk F F_lt r) (hy : y = mk F F_lt s) : HEq x y :=
+  HEq_eq_mk_eq F F_lt h e r.2 (e ▸ s.2) hx hy
+
+end
+
+lemma mk_congr {i : ι} (x y : ofClass (F i)) (h : x = y) : mk F F_lt x = mk F F_lt y :=
+  congrArg (mk F F_lt) h
+
+lemma sound [Preorder ι] [IsFiltration F F_lt] {i : ι} (x y : ofClass (F i)) :
+    x ≈ y → mk F F_lt x = mk F F_lt y :=
+  Quotient.sound
+
+@[simp]
+lemma exact [Preorder ι] [IsFiltration F F_lt] {i : ι} (x y : ofClass (F i)) :
+    mk F F_lt x = mk F F_lt y → x ≈ y :=
+  Quotient.exact
```

Comment:
As you're defining `mk ` as an `abbrev`, I'd argue you don't need any of these lemmas.